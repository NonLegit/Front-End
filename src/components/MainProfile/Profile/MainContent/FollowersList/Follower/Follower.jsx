import { Typography } from '@mui/material';
import { useEffect, useState } from 'react';
import { followRequest } from '../../../../mainProfileServer';
import {
  FollowerBox, FollowerPhoto, FollowingButton, LinkToProfile,
} from './styles';

/**
 * Follower
 *
 * @component Follower
 * @property {object} follower - entity
 * @property {object} handleClickFollow - toggle state

 * @returns {React.Component} Follower
 */
function Follower({ follower }) {
  const [following, setFollowing] = useState(undefined);
  const [isFollowedUi, setIsFollowedUi] = useState(undefined);

  useEffect(() => {
    setFollowing(follower.isFollowed);
    setIsFollowedUi(follower.isFollowed);
  }, [follower]);

  followRequest(follower.userName, following, () => { setIsFollowedUi((prev) => !prev); });

  const handleClickFollow = () => {
    setFollowing((prev) => !prev);
  };

  return (
    <FollowerBox>
      <LinkToProfile>
        <FollowerPhoto
          alt={follower.userName}
          src={follower.profilePicture}
          variant="square"
        />
        <Typography>
          {follower.userName}
        </Typography>

      </LinkToProfile>
      {isFollowedUi ? <FollowingButton onClick={handleClickFollow}>Following</FollowingButton> : <FollowingButton notfollowing="true" onClick={handleClickFollow}>Follow</FollowingButton>}
    </FollowerBox>
  );
}

export default Follower;
