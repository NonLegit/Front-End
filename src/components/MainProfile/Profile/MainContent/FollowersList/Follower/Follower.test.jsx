import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Follower from './Follower';

test('test snapshot', async () => {
  const props = {
    follower: {
      isFollowed: true,
    },
  };

  const tree = renderer.create(
    <Router>
      <Follower {...props} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
