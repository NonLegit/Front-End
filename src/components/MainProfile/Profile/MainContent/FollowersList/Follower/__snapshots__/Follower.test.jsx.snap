// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-15bt934"
>
  <a
    className="css-1lqd7h6"
    href="/"
    onClick={[Function]}
  >
    <div
      className="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-1jorvh2-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    />
  </a>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-b5yj5k-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    notfollowing="true"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Follow
  </button>
</div>
`;
